apps/coscrad-frontend/src/components/commands/command-panel.tsx

```diff
@@ -1,27 +1,42 @@
-import { AggregateCompositeIdentifier, ICommandFormAndLabels } from '@coscrad/api-interfaces';
+import {
+    AggregateCompositeIdentifier,
+    AggregateType,
+    ICommandFormAndLabels,
+    isAggregateType,
+} from '@coscrad/api-interfaces';
 import { useState } from 'react';
+import { useLoadableGeneratedId } from '../../store/slices/id-generation';
 import { DynamicForm } from '../dynamic-forms/dynamic-form';
 import { useFormState } from '../dynamic-forms/form-state';
+import { ErrorDisplay } from '../error-display/error-display';
+import { Loading } from '../Loading';
 import { CommandButton } from './command-button';
 
 export const INDEX_COMMAND_CONTEXT = 'index';
 
-type CommandContext = typeof INDEX_COMMAND_CONTEXT | AggregateCompositeIdentifier;
+export type CommandContext = AggregateType | AggregateCompositeIdentifier;
 
 interface CommandPanelProps {
     actions: ICommandFormAndLabels[];
     // Remove this if it isn't used in ID generation
     commandContext: CommandContext;
 }
 
-export const CommandPanel = ({ actions }: CommandPanelProps) => {
+export const CommandPanel = ({ actions, commandContext }: CommandPanelProps) => {
     const [selectedCommandType, setSelectedCommandType] = useState<string>(null);
 
     const [formState, updateForm] = useFormState();
 
+    const { isLoading, errorInfo, data: generatedId } = useLoadableGeneratedId();
+
     // Do not render if there are no available actions
     if (actions.length === 0) return null;
 
+    // TODO Use one of our helpers for this
```

Comment:
Give this a try.

Reply:
[Here](https://www.pivotaltracker.com/story/show/184107132) is a story for that.